hal/halx86/generic/dma.c

```diff
@@ -481,7 +481,11 @@ HalpDmaAllocateChildAdapter(IN ULONG NumberOfMapRegisters,
                                         FILE_READ_DATA | FILE_WRITE_DATA,
                                         IoAdapterObjectType,
                                         KernelMode);
-    if (!NT_SUCCESS(Status)) return NULL;
+    if (!NT_SUCCESS(Status))
```

Comment:
This failure mode literally can't happen, so it's annoying that we have to have code for it at all. If you wanted to change this to a plain `ObReferenceObject`, I'd be okay with that.

Reply:
Sure. Then, it seems it could be moved to (before/)after `ZwClose()` or even before `return`. Where do you prefer to have it?

Reply:
It can't happen after `ZwClose`, since the point is to prevent `ZwClose` from destroying the adapter. But moving it after the `ObInsertObject` simplifies error handling, so that seems worth doing, yes.

Reply:
Right. Done.